src/vs/editor/contrib/inlineCompletions/browser/singleTextEdit.ts

```diff
@@ -248,3 +247,58 @@ function smartDiff(originalValue: string, newValue: string, smartBracketMatching
 
 	return new LcsDiff({ getElements: () => elements1 }, { getElements: () => elements2 }).ComputeDiff(false).changes;
 }
+
+/**
+ * Given some text edits, this function finds the new ranges of the editted text post application of all edits.
+ * Assumes that the edit ranges are disjoint
+ * @param edits edits applied
+ * @returns new ranges post edits for every edit
+ */
+export function getNewRanges(edits: SingleTextEdit[]): Range[] {
+
+	if (edits.length === 0) {
+		return [];
+	}
+
+	const sortIndices = Array.from(edits.keys()).sort((a, b) =>
+		Range.compareRangesUsingStarts(edits[a].range, edits[b].range)
+	);
+	const ranges: Range[] = [];
+	let previousEditEndLineNumber = 0;
+	let positionOffset = new Position(0, 0);
+
+	for (const index of sortIndices) {
+		const edit = edits[index];
+		const splitText = splitLines(edit.text!);
+		const rangeStart = edit.range.getStartPosition().delta(
```

Comment:
You can use `addPositions` for this

Reply:
I wonder if that would make the code cleaner given that when I use the delta method, for the column offset I use a ternary condition as follows? 

edit.range.startLineNumber === previousEditEndLineNumber ? positionOffset.column : 0